store/migration/dev/20220419000000__sheet_stargazer.sql

```diff
@@ -0,0 +1,8 @@
+-- sheet_stargazer table stores the sheet stargazers.
+CREATE TABLE sheet_stargazer (
+    sheet_id INTEGER NOT NULL REFERENCES sheet (id),
```

Comment:
We may want to be consistent with other tables to start a table with auto-increment IDs.
```
CREATE TABLE sheet_stargazer (
    id SERIAL PRIMARY KEY,
    sheet_id INTEGER NOT NULL REFERENCES sheet (id),
    principal_id INTEGER NOT NULL REFERENCES principal (id),
    type(or something else) TEXT NOT NULL CHECK (type IN ('STARRED', 'UNSTARRED')) DEFAULT 'STARRED'
);

CREATE UNIQUE INDEX idx_sheet_stargazer_unique_sheet_id_principal_id ON sheet_stargazer(sheet_id, principal_id);
```

Reply:
I am thinking of a different structure to solve both starred and pinned.

CREATE TABLE sheet_organizer (
....
bookmarked BOOLEAN
pinned BOOLEAN
)

thoughts @boojack @d-bytebase ?

Reply:
I think starred is a type of pinned. We can show starred sheets firstly in UI, so they are a bit of same.

Reply:
> I think starred is a type of pinned. We can show starred sheets firstly in UI, so they are a bit of same.

But if we need to pin the starred sheets, then we need `pinned`.

Reply:
star and pin can be different. One can star a lot of sheets, while she may only pin a few of them. The difference is subtle so we do not need to add pin for now. Meanwhile, I would still vote for the starred boolean instead of a "type" field.

I am not seeing a 3rd value that is mutually exclusive from STARRED and UNSTARRED.